File: mobile/lib/models/part.dart
class Part {
  final int id;
  final String name;
  final String? reference;
  final String unit;
  final int quantityOnHand;

  Part({
    required this.id,
    required this.name,
    this.reference,
    required this.unit,
    required this.quantityOnHand,
  });

  factory Part.fromJson(Map<String, dynamic> json) {
    return Part(
      id: json['id'],
      name: json['name'],
      reference: json['reference'],
      unit: json['unit'],
      quantityOnHand: json['quantity_on_hand'],
    );
  }
}

class PartUsage {
  final Part part;
  int quantity;

  PartUsage({required this.part, this.quantity = 1});
}

File: backend/app/Models/Part.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToTenant;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Part extends Model
{
    use BelongsToTenant, HasFactory;

    protected $fillable = [
        'tenant_id',
        'name',
        'reference',
        'unit',
        'quantity_on_hand',
        'alert_threshold',
        'unit_cost',
        'supplier',
    ];

    protected $casts = [
        'unit_cost' => 'decimal:2',
    ];

    public function stockMovements()
    {
        return $this->hasMany(StockMovement::class);
    }

    public function isBelowThreshold(): bool
    {
        return $this->quantity_on_hand <= $this->alert_threshold;
    }
}
